packages/mask/src/plugins/Savings/protocols/AAVEProtocol.ts

```diff
@@ -0,0 +1,311 @@
+            // APY and APR are returned here as decimals, multiply by 100 to get the percents
+            const apr = liquidityRate / RAY
+            this.apr = apr.toFixed(2)
+            return apr.toFixed(2)
+        } catch (error) {
+            console.log('AAVE `getApr()` error', error)
+            // Default APR
+            this.apr = '0.17'
+            return '0.17'
+        }
+    }
+
+    public async getBalance(chainId: ChainId, web3: Web3, account: string) {
+        try {
+            const subgraphUrl = getSavingsConstants(chainId ?? ChainId.Kovan).AAVE_SUBGRAPHS || ''
+            const reserveBody = JSON.stringify({
+                query: `{
+                reserves (where: {
+                    underlyingAsset: "${getSavingsConstants(chainId ?? ChainId.Kovan).AAVE || ZERO_ADDRESS}"
+                }) {
+                    id
+                    name
+                    underlyingAsset            
+                }
+            }`,
+            })
+
+            const reserveResponse = await fetch(subgraphUrl, {
+                method: 'POST',
+                headers: { 'Content-Type': 'application/json' },
+                body: reserveBody,
+            })
+            const fullResponse = await reserveResponse.json()
+            const reserveId = fullResponse.data.reserves[0].id
+
+            // Get User Reserve
+            const userReserveBody = JSON.stringify({
+                query: `{
+                    userReserves(where: { 
+                        user: "${account}",
+                        reserve: "${reserveId}"
+                        
+                        }) {
+                      id
+                      scaledATokenBalance
+                      currentATokenBalance
+                      reserve{
+                        id
+                        symbol
+                        underlyingAsset
+                        decimals
+                      }
+                      user {
+                        id
+                      }
+                    }
+                  }`,
+            })
+
+            const userReserveResponse = await fetch(subgraphUrl, {
+                method: 'POST',
+                headers: { 'Content-Type': 'application/json' },
+                body: userReserveBody,
+            })
+            const userResponse = await userReserveResponse.json()
+
+            const balance = userResponse.data.userReserves[0].currentATokenBalance
+            this.balance = new BigNumber(balance || '0')
+            return this.balance
+        } catch (error) {
+            console.log('AAVE `getBalance()` error', error)
+            this.balance = new BigNumber('0')
+            return this.balance
+        }
+    }
+
+    public async depositEstimate(account: string, chainId: ChainId, web3: Web3, value: BigNumber.Value) {
+        try {
+            const lPoolAdressProviderContract = createContract<AaveLendingPoolAddressProvider>(
```

Comment:
Refactor the create contract code to be a private method if you need it multiple times.

Reply:
Done